pkg/sfu/downtrack.go

```diff
@@ -694,13 +695,13 @@ func (d *DownTrack) CloseWithFlush(flush bool) {
 }
 
 func (d *DownTrack) SetMaxSpatialLayer(spatialLayer int32) {
+	d.logger.Infow("setting max spatial layer", "layer", spatialLayer)
 	changed, maxLayers, currentLayers := d.forwarder.SetMaxSpatialLayer(spatialLayer)
 	if !changed {
 		return
 	}
 
-	if d.onMaxLayerChanged != nil && d.kind == webrtc.RTPCodecTypeVideo &&
-		maxLayers.SpatialGreaterThanOrEqual(currentLayers) && d.bound.Load() {
+	if d.onMaxLayerChanged != nil && d.kind == webrtc.RTPCodecTypeVideo && maxLayers.SpatialGreaterThanOrEqual(currentLayers) {
```

Comment:
I think this is where the issue is with stress test failure. Subscription update happening before bind. We had an update in Bind callback, but that was causing a different race and hence removed.